runtime/lib/ttnn/program.cpp

```diff
@@ -38,15 +40,26 @@ class ProgramExecutor {
   ProgramExecutor(const TensorMap &liveTensors,
                   const std::unordered_set<uint32_t> &programInputs,
                   const std::unordered_set<uint32_t> &programOutputs,
+                  const std::unordered_map<std::string, Tensor> &programGoldens,
                   ::ttnn::MeshDevice *meshDevice)
       : context(ProgramContext(liveTensors, programInputs, programOutputs,
-                               meshDevice)) {}
+                               programGoldens, meshDevice)) {}
 
   void execute(const ::tt::target::ttnn::Program *program) {
     for (const ::tt::target::ttnn::Operation *op : *program->operations()) {
       LOG_DEBUG(LogType::LogRuntimeTTNN,
                 "Executing operation: ", op->debug_info()->c_str());
       runOperation(op);
+      if (auto callback = debug::Hooks::get().getOperatorCallback(); callback) {
```

Comment:
Can we move this into its own function. Something like `runCallback`

Reply:
fixed